docs/RFD-JobWorkerService.md

```diff
@@ -0,0 +1,366 @@
+
+* **func (\*Job) Stop() error** - attempt to gracefully shutdown the process.
+`Stop` will do nothing if the process has already completed or been killed.
+
+### API 
+The API is nothing more than just a wrapper around Library, but main API responsibility provide mTLS Authorization and using Library
+functionality via [gRPC](https://grpc.io/)
+
+#### Following functions considered to be used for API Server.
+
+* `Start` New Job
+  * verify user credentials via mTLS Authorization
+  * create new instance of the `Job` and assign new UUID 
+  * store `job` in sync.Map using UUID as key
+  * start command execution
+  * return UUID to the client
+* `Status` of the Job
+  * return current status of the `Job`
+* `Stream` latest Job output
+  * lookup `Job` by provided UUID
+  * request instance of `OutputReader` using library
+  * send steam output to the client/cli
+* `Stop` Job
+  * lookup `Job` by provided UUID
+  * call `Stop` function if `Job` was found, otherwise return error
+  
+  > **Not any job and it output removed from then server**. So, depend on the load and size of total output it would come to OOM situation sooner or later and [panic](https://go.dev/blog/defer-panic-and-recover).
+  > Probably we need to consider some API or process to clean completed jobs based on size, date of creation, frequency of usage or etc.
+  
+
+#### Security
+
+The client and API communicate via mTLS using [TLS 1.3 ](https://datatracker.ietf.org/doc/html/rfc8446) as the minimum version. The following cipher suites are supported:
+
+* TLS_AES_256_GCM_SHA384
+* TLS_CHACHA20_POLY1305_SHA256
+* TLS_AES_128_GCM_SHA256
+
+> Note: only 3 cipher suites enabled openssl by default: https://wiki.openssl.org/index.php/TLS1.3, and following 2 
+> below also can be used, but need to explicitly enabled: 
+>* TLS_AES_128_CCM_8_SHA256
+>* TLS_AES_128_CCM_SHA256
+
+> Since there are no requirements for older versions of TLS to be supported, TLS v1.3 will be made a minimum.
+
+#### Authorization
+All verification should be done on API level and API treats the client's common name (CN) in the certificate as the user's identity.
+Client not able interact with a job not created by them in any way (Stream, query Status, Stop for another user's job is not allowed).
+
+API not cache any user authorization data and such data should be provided for every API call.
+
+#### Proposed Protobuf
+Following [protobuf](https://protobuf.dev/programming-guides/proto3/) proposed for the client/server:
+
+```protobuf
+  syntax="proto3";
+
+  option go_package = "<package_name>/proto";
+
+  package proto;
+
+  service JobWorker {
+    rpc Start(JobCreateRequest) returns (JobResponse) {}
+    rpc Status(JobRequest) returns (JobStatusResponse) {}
+    rpc Stream(JobRequest) returns (stream OutputResponse) {}
+    rpc Stop(JobRequest) returns (JobStatusResponse) {}
+  }
+
+  // requests
+  message JobCreateRequest {
+    double  cpu = 1;
+    int64   memBytes = 2;
+    int64   ioBytesPerSecond = 3;
```

Comment:
For simplify you can hardcore the limits on the server side.

Reply:
@smallinsky my plan is set _default_ values in cli level and keep API clean. If team prefer have them hardcode on server side - I can do this. What do you think ?

<img width="978" alt="Screenshot 2024-08-26 at 10 49 42 AM" src="https://github.com/user-attachments/assets/afa7c3e0-ae18-4cbb-9efd-470b864dd0de">

Reply:
That should works 👍